server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java

```diff
@@ -390,6 +391,28 @@ public void testDuelingStrictParsing() {
 
         assertSameDate("2012-W31-5", "strict_weekyear_week_day");
         assertParseException("2012-W1-1", "strict_weekyear_week_day");
+//failing below
+        assertSameDate("2015-01-04T00:00Z", "strict_date_optional_time||epoch_millis");
```

Comment:
can you just use `strict_date_optional_time` and remove `epoch_millis`? Also, can yo move this to the other `strict_date_optional_time` checks?

Can you also move this `strict_date_optional_time` checks to the others in this method?

Reply:
absolutely - will clean this up